docs/openid.md

```diff
@@ -89,7 +90,51 @@ oidc_config:
        localpart_template: "{{ user.name }}"
        display_name_template: "{{ user.name|capitalize }}"
 ```
+### [Keycloak][keycloak-idp]
 
+[Keycloak][keycloak-idp] is an opensource IdP maintained by Red Hat. 
+
+Follow the [Getting Started Guide](https://www.keycloak.org/getting-started) to install Keycloak and set up a realm.
+
+1. `Clients -> Create`
+
+2. Fill Fields
+
+| Field | Value |
+|-----------|-----------|
+| Client ID | `synapse` |
+| Client Protocol | `openid-connect` |
+
+3. `Save`
+
+4. Fill Fields
+
+| Field | Value |
+|-----------|-----------|
+| Client ID | `synapse` |
+| Enabled | `On` |
+| Client Protocol | `openid-connect` |
+| Access Type | `confidential` |
+| Valid Redirect URIs | `[synapse public baseurl]/_synapse/oidc/callback` |
+
+5. Save
+6. Credential Tab
+
+| Field | Value |
+|-------|-------|
+| Client Authenticator | `Client ID and Secret` |
+
+7. Regenerate Secret
+8. Copy Secret
+
+```yaml
+oidc_config:
+   enabled: true
+   issuer: "https://127.0.0.1:8443/auth/realms/{realm_name}"
+   client_id: "synapse"
+   client_secret: "copy secret from Step 8"
```

Comment:
It might be best to just say "from above" instead of "step 8" in case the number of steps changes!

Reply:
The client secret is within the crendential tab

Reply:
https://github.com/matrix-org/synapse/pull/7659/commits/015444e40a490196481f68384ec1066a19f1195e#diff-de854b597751e2f32829a3fb90525167R134